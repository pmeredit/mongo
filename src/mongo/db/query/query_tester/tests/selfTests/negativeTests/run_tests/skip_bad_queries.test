skip_bad_queries
negative_test
test.coll as coll

// This test should execute successfully and be numbered 0.
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toString": "$value" } } } ], cursor: {} }

// This test should fail and be skipped.
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toDate": "$value" } } } ], cursor: {} }

// This test should execute successfully and be numbered 2.
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toLong": "$_id" } } } ], cursor: {} }
