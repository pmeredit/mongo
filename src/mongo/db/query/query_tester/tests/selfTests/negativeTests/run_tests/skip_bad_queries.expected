skip_bad_queries
negative_test
test.coll as coll

// This test should execute successfully and be numbered 0.
0
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toString": "$value" } } } ], cursor: {} }
[
{"_id":0,"projected":"a","value":"a"},
{"_id":1,"projected":"b","value":"b"}
]

// This test should execute successfully and be numbered 2.
2
:sortFull { "aggregate" : "coll", "pipeline" : [ { "$addFields" : { "projected" : { "$toLong": "$_id" } } } ], cursor: {} }
[
{"_id":0,"projected":0,"value":"a"},
{"_id":1,"projected":1,"value":"b"}
]
